internal/app/singularity/cache_list_linux.go

```diff
@@ -157,29 +166,59 @@ func ListSingularityCache(cacheListTypes []string, listAll bool) error {
 		}
 	}
 
-	fmt.Printf("%-22s %-22s %-16s %s\n", "NAME", "DATE CREATED", "SIZE", "TYPE")
+	var containerCount int
+	var containerSpace int64
+	var blobCount int
+	var blobSpace int64
+	var totalSpace int64
+
+	if !cacheListSummery {
+		fmt.Printf("%-22s %-22s %-16s %s\n", "NAME", "DATE CREATED", "SIZE", "TYPE")
+	}
 
 	if libraryList || listAll {
-		if err := listLibraryCache(); err != nil {
+		libraryCount, librarySize, err := listLibraryCache(cacheListSummery)
+		if err != nil {
 			return err
 		}
+		containerCount += libraryCount
+		containerSpace += librarySize
 	}
 	if ociList || listAll {
-		if err := listOciCache(); err != nil {
+		ociCount, ociSize, err := listOciCache(cacheListSummery)
+		if err != nil {
 			return err
 		}
+		containerCount += ociCount
+		containerSpace += ociSize
 	}
 	if blobList || listAll {
-		if err := listBlobCache(true); err != nil {
+		blobsCount, blobsSize, err := listBlobCache(true)
+		if err != nil {
+			return err
+		}
+		blobCount = blobsCount
+		blobSpace = blobsSize
+	} else {
+		blobsCount, blobsSize, err := listBlobCache(false)
+		if err != nil {
 			return err
 		}
-		// dont list blob summary after listing all blobs
-		listBlobSum = false
+		blobCount = blobsCount
+		blobSpace = blobsSize
 	}
-	if listBlobSum {
-		if err := listBlobCache(false); err != nil {
+
+	if !listAll || cacheListSummery {
+		totalSpace = containerSpace + blobSpace
+		realTotalSpace, err := findSize(totalSpace)
+		if err != nil {
 			return err
 		}
+		realContainerSpace, _ := findSize(containerSpace)
+		realBlobSpace, _ := findSize(blobSpace)
+		fmt.Printf("\nThere %v containers using: %v, %v oci blob file(s) using %v of space.\n", containerCount, realContainerSpace, blobCount, realBlobSpace)
```

Comment:
update this line to:

```
fmt.Printf("\nThere are %v container(s) using: %v, %v oci blob file(s) using %v of space.\n", containerCount, realContainerSpace, blobCount, realBlobSpace)
    
```

Let's enforce the "There are x container(s)" message to avoid issue when there's a single container, please do change where applicable on other functions :+1: